internal/pkg/controllers_metrics/base_metrics.go

```diff
@@ -0,0 +1,31 @@
+package controllers_metrics
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type BaseMetrics struct {
+	ReconciliationsTotal       prometheus.Counter
+	ReconciliationsFailedTotal prometheus.Counter
+	ClientErrorsTotal          prometheus.Counter
+}
+
+func NewBaseMetrics(reg prometheus.Registerer, component string) *BaseMetrics {
+	return &BaseMetrics{
+		ReconciliationsTotal: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+			Name: fmt.Sprintf("thanos_operator_%s_reconciliations_total", component),
```

Comment:
I think the idea was to just make it a thanos_operator_reconcilations_total metric with a controller label. That way we don't really need an interface. And we can just register the metrics once somewhere, and just use WithLabel increments in the controllers themselves. 🙂